src/sentry/integrations/github/integration.py

```diff
@@ -666,55 +666,106 @@ def get_organization_config(self) -> list[dict[str, Any]]:
         config = self._get_organization_config_default_values()
 
         if features.has("organizations:integrations-github-project-management", self.organization):
-            config.insert(
-                0,
-                {
-                    "name": self.outbound_status_key,
-                    "type": "choice_mapper",
-                    "label": _("Sync Sentry Status to Github"),
-                    "help": _(
-                        "When a Sentry issue changes status, change the status of the linked ticket in Github."
-                    ),
-                    "addButtonText": _("Add Github Project"),
-                    "addDropdown": {
-                        "emptyMessage": _("All projects configured"),
-                        "noResultsMessage": _("Could not find Github project"),
-                        "items": [],  # Populated with projects
-                    },
-                    "mappedSelectors": {},
-                    "columnLabels": {
-                        "on_resolve": _("When resolved"),
-                        "on_unresolve": _("When unresolved"),
-                    },
-                    "mappedColumnLabel": _("Github Project"),
-                    "formatMessageValue": False,
-                },
-            )
-            try:
-                # Fetch all repositories and add them to the config
-                repositories = self.get_client().get_repos()
-
-                # Format repositories for the dropdown
-                formatted_repos = [
-                    {"value": repository["full_name"], "label": repository["name"]}
-                    for repository in repositories
-                    if not repository.get("archived")
-                ]
-                config[0]["addDropdown"]["items"] = formatted_repos
+            if features.has(
+                "organizations:integrations-external-projects-async-lookup", self.organization
+            ):
+                # Async lookup for integration external projects in the frontend
+                # Get currently configured external projects to display their labels
+                current_repo_items = []
+                external_projects = IntegrationExternalProject.objects.filter(
+                    organization_integration_id=self.org_integration.id
+                )
```

Comment:
The nesting + diff makes this a little hard to compare. In the future, could you split the new behavior into a helper function please?